config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleMetadataBuilder.java

```diff
@@ -9,42 +9,97 @@
 import com.ca.apim.gateway.cagatewayconfig.beans.Encass;
 import com.ca.apim.gateway.cagatewayconfig.beans.GatewayEntity;
 import com.ca.apim.gateway.cagatewayconfig.beans.Policy;
+import com.ca.apim.gateway.cagatewayconfig.util.IdGenerator;
+import org.apache.commons.lang3.StringUtils;
 
+import javax.inject.Inject;
 import javax.inject.Singleton;
-import java.util.*;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 import java.util.stream.Collectors;
 
-import static com.ca.apim.gateway.cagatewayconfig.bundle.builder.BuilderConstants.FILTER_ENV_ENTITIES;
+import static com.ca.apim.gateway.cagatewayconfig.bundle.builder.BuilderConstants.*;
 
 @Singleton
 public class BundleMetadataBuilder {
+    private final IdGenerator idGenerator;
 
+    @Inject
+    BundleMetadataBuilder (IdGenerator idGenerator) {
+        this.idGenerator = idGenerator;
+    }
+
+    /**
+     * Builds bundle metadata for annotated entities.
+     *
+     * @param annotatedBundle
+     * @param annotatedEntity
+     * @param dependentEntities
+     * @param projectName
+     * @param projectGroupName
+     * @param projectVersion
+     * @return bundle metadata
+     */
     public BundleMetadata build(final AnnotatedBundle annotatedBundle,
                                 final AnnotatedEntity<? extends GatewayEntity> annotatedEntity,
```

Comment:
Can you remove this argument? `AnnotatedBundle` contains the annotatedEntity so we can use that.